mir-ci/mir_ci/tests/test_mir_flutter_app.py

```diff
@@ -0,0 +1,439 @@
+    ELSE IF    $preset == "RIGHT"
+        ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * -4)
+    ELSE IF    $preset == "BOTTOM_LEFT"
+        ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * -3)
+    ELSE IF    $preset == "BOTTOM"
+        ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * -2)
+    ELSE IF    $preset == "BOTTOM_RIGHT"
+        ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * -1)
+    ELSE IF    $preset == "CENTER"
+        ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * 0)
+    ELSE IF    $preset == "CUSTOM"
+        ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * 1)
+    ELSE
+        Fail    Unexpected preset: ${preset}
+    END
+    Walk Pointer To ${pos}
+    Click And Sleep
+
+Move Main Window To The Top Left Of The Output
+    ${pos}=    Move Pointer To ${GROUP_NEW_WINDOW}
+    ${pos}=    Displace ${pos} By (0, -77)
+    Walk Pointer To ${pos}
+    Press LEFT Button
+    Walk Pointer To (600, 0)
+    Release LEFT Button
+
+Set Top Left Custom Positioner
+    Move Pointer To ${BUTTON_CUSTOM_PRESET}
+    Click And Sleep
+    Walk Pointer To ${BUTTON_SET_DEFAULTS}
+    Click And Sleep
+    # Set parent anchor to 'topLeft'
+    ${pos}=    Walk Pointer To ${DROPDOWN_PARENT_ANCHOR_LABEL}
+    Click And Sleep
+    ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * 4)
+    Walk Pointer To ${pos}
+    Click And Sleep
+    # Set child anchor to 'bottomRight'
+    ${pos}=    Walk Pointer To ${DROPDOWN_CHILD_ANCHOR_LABEL}
+    Click And Sleep
+    ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * 5)
+    Walk Pointer To ${pos}
+    Click And Sleep
+    Walk Pointer To ${BUTTON_APPLY}
+    Click And Sleep
+"""
+
+
+@pytest.mark.parametrize("server", servers(ServerCap.MIR_FLUTTER_APP))
+@pytest.mark.deps(
+    pip_pkgs=(
+        ("robotframework~=6.1.1", "robot"),
+        ("rpaframework", "RPA"),
+        ("rpaframework-recognition", "RPA.recognition"),
+    ),
+)
+class TestMirFlutterApp:
+    async def test_mir_flutter_app(self, robot_log, server, tmp_path) -> None:
+        extensions = VirtualPointer.required_extensions + ScreencopyTracker.required_extensions
+        server_instance = DisplayServer(server, add_extensions=extensions)
+
+        robot_test_cases = dedent(
+            """\
+            Reference App Opens
+                Sleep    2
+                VIDEO.Match    ${WINDOW_MAIN}
+
+            Regular Window Opens
+                Move Pointer To ${BUTTON_REGULAR}
+                Click And Sleep
+                VIDEO.Match    ${WINDOW_REGULAR_0_FOCUSED}
+                Close Focused Toplevel Window
+
+            Floating Regular Window Opens
+                Move Pointer To ${BUTTON_FLOATING_REGULAR}
+                Click And Sleep
+                VIDEO.Match    ${WINDOW_FLOATING_REGULAR_0_FOCUSED}
```

Comment:
When we're matching here, why do we only seem to match the titlebar and not the whole window?

Reply:
It's a small template that uniquely identifies the window. I think it's good enough because the title bar is rendered by the client. There's another reason: It would be hard to match against a window with rounded corners because the image matcher doesn't account for images with alpha channel.

Reply:
Would the title bar not also have rounded corners in that case? Or I guess it wouldn't matter because we'd only be matching a subset of the title bar?

Reply:
Exactly, we match only the rectangle containing the title text.